### PR TITLE
feat: implement new vc-pagination

### PR DESCRIPTION
**Pull request by goldenmaya**
### Jira-link:

https://virtocommerce.atlassian.net/browse/ST-5153
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.43.0-pr-849-7a7c-7a7c8e14.zip

### CODE REVIEW

File: client-app/ui-kit/components/organisms/pagination/vc-pagination.vue [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        <VcButton
+          class="vc-pagination__button vc-pagination__button--prev"
+          color="secondary"
+          variant="outline"
+          size="sm"
+          :disabled="page === 1"
+          @click="setPage(page - 1)"
+        >
+          <VcIcon name="chevron-left" />
+          <span>Previous</span>
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/organisms/pagination/vc-pagination.vue:33]: Should be localized

File: client-app/ui-kit/components/organisms/pagination/vc-pagination.vue [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+        <VcButton
+          class="vc-pagination__button vc-pagination__button--next"
+          color="secondary"
+          variant="outline"
+          size="sm"
+          :disabled="page === pages"
+          @click="setPage(page + 1)"
+        >
+          <span>Next</span>
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/organisms/pagination/vc-pagination.vue:44]: Should be localized

File: client-app/ui-kit/components/organisms/pagination/vc-pagination.stories.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { VcPagination } from "..";
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/organisms/pagination/vc-pagination.stories.ts:1]: May be add all states (1,2,3,4,5, 1...4,5,6...10) to storybook to see all possible looks in UI in one place?

File: client-app/ui-kit/components/organisms/pagination/vc-pagination.vue [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+function scrollToTop() {
+  if (props.scrollTarget) {
+    const topPosition = props.scrollTarget.getBoundingClientRect().top + window.scrollY - props.scrollOffset;
+    window.scrollTo({ top: topPosition, behavior: "smooth" });
+  }
+}
+
+const setPage = (page: number) => {
+  if (page) {
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/organisms/pagination/vc-pagination.vue:118]: Unnecessary check or wrong type: if page is number check is redundand, if i can be null or undefined - type is wrong.

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: client-app/ui-kit/components/organisms/pagination/vc-pagination.vue [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+    for (let i = props.page - 2; i <= props.page + 2; i++) {
+      pages.push(i);
+    }
+
+    pages.push(0);
+    pages.push(props.pages);
+  }
+
+  return pages;
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/organisms/pagination/vc-pagination.vue:107]: It's better to use NaN or special constant

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

**Review by asvishnyakov — APPROVED** [review]

**Review by NaMax66 — APPROVED** [review]

**Pull request merged**